docs/proposals/groups.md

```diff
@@ -0,0 +1,94 @@
+```
+// OrgGroup contains the set of Users that belong to a given OrgGroup.  The name of a OrgGroup is required to 
+// to be in the form of "OrganizationName:OrgGroupName".
+type OrgGroup struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	// Users is a list of Users that are members of the OrgGroup
+	Users util.StringSet
+}
+```
+
+A reverse lookup index will be built in memory (similar to the project cache) to be allow the authenticator to make fast lookups for OrgGroup membership determination.  A valid OrgGroup.Name can never be used as a ClusterGroup.Name and vice-versa.  This detail is important later on for authorization parity with kube and backwards compatibility with v3 of OpenShift.
+
+
+## Organizations
+An Organization is a cluster scoped API object that references Users and has subresources for accessing and modifying Organization scoped OrgGroups.
+```
+// Organization contains the set of Users that are members of the Organization.
+type Organization struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	// AdminUsers is a list of Users that can administer this Organization.  By default a policy rule exists
+	// that allows all Users to inspect and modify OrgGroups of Organizations that they are an Admin for.
+	// Admins may NOT modify Organization membership.
+	AdminUsers util.StringSet
+	// AdminClusterGroups is a list of ClusterGroups that can administer this Organization.  By default a policy rule exists
+	// that allows all Users to inspect and modify OrgGroups of Organizations that they are an Admin for.
+	// Admins may NOT modify Organization membership.
+	AdminClusterGroups util.StringSet
+
+	// Members is a list of Users that are members of the Organization.  Only members may be assigned
+	// to OrgGroups.
+	Members util.StringSet
+}
+```
+
+The Organization has a list of admins.  A default policy rule is bound to all-authenticated users that allows them inspect and modify OrgGroups of Organizations that they are an Admin for (whether via User or ClusterGroup).  Having the policy rule described and bound in this way makes it easier to manage mutiple organizations without having a proliferation of roles and rolebindings.
+
+The Organization has a list members.  Only users from this list may be assigned to OrgGroups.  This prevents an overzelous organization admin from creating OrgGroups made up on non-members.  By default, an organization admin does not have rights to change Organization membership.
+
+*****TBD*****: How are users added to the Organization???  Being able to modify the OrgGroup structure of known members seem logically distinct from being able to force users into an Organization.  I wouldn't expect the same person to do both.
```

Comment:
> Or just a group? Most companies are going to make organizations map an LDAP group. The rest are going to be self adding (admin can add members).

It makes logic about who can be a member of an OrgGroup move involved and given current default permissions makes it impossible for an OrgAdmin to determine who is a member (they can't see groups), but I don't have any real objection to allowing ClusterGroups.  We could add a subresource to Organizations to flatten the list of users and grant OrgAdmins permission to see it.